cache/clock_cache.cc

```diff
@@ -588,14 +593,23 @@ Status ClockCacheShard::Insert(const Slice& key, uint32_t hash, void* value,
                                size_t charge,
                                void (*deleter)(const Slice& key, void* value),
                                Cache::Handle** out_handle,
-                               Cache::Priority /*priority*/) {
+                               Cache::Priority /*priority*/,
+                               bool charge_internal_usage) {
   CleanupContext context;
   HashTable::accessor accessor;
   char* key_data = new char[key.size()];
   memcpy(key_data, key.data(), key.size());
   Slice key_copy(key_data, key.size());
-  CacheHandle* handle = Insert(key_copy, hash, value, charge, deleter,
-                               out_handle != nullptr, &context);
+  if (charge_internal_usage) {
+#ifdef ROCKSDB_MALLOC_USABLE_SIZE
+    charge += malloc_usable_size(static_cast<void*>(key_data));
```

Comment:
The PR description talks only about the BlockHandle overhead but here it seems that there is an overhead of key as well.

Reply:
updated.